packages/graphiql-react/src/editor/context.tsx

```diff
@@ -423,20 +441,45 @@ export function EditorContextProvider(props: EditorContextProviderProps) {
     [onTabChange, setEditorValues, storeTabs],
   );
 
+  const closeTabConfirmation = useCallback<
+    EditorContextType['closeTabConfirmation']
+  >(
+    async index => {
+      if (confirmCloseTab) {
+        const confirmation = await confirmCloseTab(index);
+        return confirmation;
+      }
+      return true;
+    },
+    [confirmCloseTab],
+  );
+
   const closeTab = useCallback<EditorContextType['closeTab']>(
-    index => {
-      setTabState(current => {
-        const updated = {
-          tabs: current.tabs.filter((_tab, i) => index !== i),
-          activeTabIndex: Math.max(current.activeTabIndex - 1, 0),
-        };
-        storeTabs(updated);
-        setEditorValues(updated.tabs[updated.activeTabIndex]);
-        onTabChange?.(updated);
-        return updated;
-      });
+    async index => {
+      if (await closeTabConfirmation(index)) {
+        if (index === tabState.activeTabIndex) {
+          executionContext?.stop();
```

Comment:
this will never be called, because we can't execution context inside the editor context, because we already use editor context inside execution context. to reproduce it simple change 
```suggestion
          executionContext!.stop();
```

and use `useExecutionContext({ notNull: true });`